### PR TITLE
VACMS-15240 Stand up running next-build server per tugboat instance

### LINKED ISSUE #15240: Spike: Stand up running Next.js server per Tugboat instance
## Requirements

Each CMS Tugboat instance should have a domain added that points to a running Next Build server running for preview purposes. 

All these AC are for the working branch for this ticket. We are **not** expecting to merge this at this stage.
```[tasklist]
### Acceptance criteria
- [ ] Upon spinning up, Tugboat checks out next-build code and starts up a Next.js server on that Tugboat instance.
- [ ] That Next.js instance can be reached at a Tugboat domain with the prefix `next-`
```

## Background & implementation details

CMS Tugboat configuration:
* https://github.com/department-of-veterans-affairs/va.gov-cms/tree/main/.tugboat
  * Note in particular config.yml and vhost-*.yml

Information about our use of Tugboat for CMS:
* https://github.com/department-of-veterans-affairs/va.gov-cms/blob/main/READMES/tugboat.md

Devops info about Tugboat CMS implementation:
* https://github.com/department-of-veterans-affairs/va.gov-cms/blob/main/READMES/devops/tugboat.md
* https://github.com/department-of-veterans-affairs/va.gov-cms/blob/main/READMES/devops/tugboat-on-premise-installation.md

Tugboat's own documentation
* https://docs.tugboatqa.com/

VA Slack Tugboat information and support channel: 
* https://dsva.slack.com/archives/C01A35JDH88
```[tasklist]
### Implementation tasks
- [ ] Use this if it helps you or feel free to delete.
```

### PR DESCRIPTION
**Pull request by KitLykos**
## Description

Closes #15240. 

- Adds scripts to clone & install, build, and start the next-build repo as part of the cms tugboat environments. Mostly based on how content-build is included on tugboat currently through `.web`
- Creates apache config for tugboat in order to route requests with `next-*` to the running server

## Testing done
in the generated tugboat links below:
- visit the `next-*` prefix link and see that the 404 page is coming from next-build
- visit `/api/preview` and see a server error, because the config to communicate with drupal isn't there yet

- all existing generated tugboat links should resolve & behave as expected (`web-*`, `storybook-*`,  `cms-*`)

## Screenshots
Screenshot 2023-10-18 at 4 04 57 PM
(this is what the 404 page from next.js looks like by default)

## QA steps
- all existing generated tugboat links should resolve & behave as expected (`web-*`, `storybook-*`,  `cms-*`)

Links:

https://pr15429-jnicnrrsvgrc1z8mpnqng14egyolhboj.ci.cms.va.gov/
https://cms-jnicnrrsvgrc1z8mpnqng14egyolhboj.ci.cms.va.gov/
should resolve to the drupal login screen

https://web-jnicnrrsvgrc1z8mpnqng14egyolhboj.ci.cms.va.gov/
should show the 403 page telling you to request a front end build. when that is requested & built, it should load the front end of va.gov.

https://storybook-jnicnrrsvgrc1z8mpnqng14egyolhboj.ci.cms.va.gov/
should show the 403 page telling you to check the logs for the storybook status. 
once it's built, it should resolve to the cms version of storybook

https://next-jnicnrrsvgrc1z8mpnqng14egyolhboj.ci.cms.va.gov/
before the server is running, it should return the 403 page like the others
once the server is running, it should resolve to a next.js 404 page 

https://next-jnicnrrsvgrc1z8mpnqng14egyolhboj.ci.cms.va.gov/api/preview should resolve to the 503 page, because the configuration between drupal & next-build isn't in place per env yet

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [x] `CMS Team`
- [ ] `Public websites`
- [ ] `Facilities`
- [ ] `User support`
- [x] `Accelerated Publishing`

### CODE REVIEW

**Comment by ndouglas** [comment]
Please keep your PR in draft rather than relying on the "Do Not Merge" label 🙂

File: composer.json [hunk 1]
```diff
@@ -573,7 +573,8 @@
         "va:theme:watch": "Watch VA.gov Claro theme source and compile as needed.",
         "va:web:build": "Perform a content build.",
         "va:web:install": "Prepare the content-build project.",
-        "va:web:prepare-dotenv": "Prepare the ./web/.env file."
+        "va:web:prepare-dotenv": "Prepare the ./web/.env file.",
+		"va:next-preview": "Prepare the next preview server"
```
Comment (tjheffner) [inline composer.json:577]: this spacing is off

File: composer.json [hunk 2]
```diff
@@ -819,6 +820,21 @@
             "# Prepare the ./web/.env file.",
             "! ./scripts/should-run-directly.sh || ./scripts/web-prepare-dotenv.sh",
             "./scripts/should-run-directly.sh || ddev composer va:web:prepare-dotenv --"
+        ],
+		"va:next:install": [
```
Comment (tjheffner) [inline composer.json:824]: spacing is off for all 3 of these, the indents should match the other commands

File: composer.json [hunk 3]
```diff
@@ -819,6 +820,21 @@
...
+            "./scripts/should-run-directly.sh || ddev composer va:next:install --"
+        ],
+        "va:next:build": [
+            "# Prepare the next build server",
+			"! ./scripts/should-run-directly.sh || ./scripts/next-build.sh",
+            "./scripts/should-run-directly.sh || ddev composer va:next:build --"
+        ],
+        "va:next:start": [
+            "# Start the next build server",
+			"! ./scripts/should-run-directly.sh || yarn start",
```
Comment (tjheffner) [inline composer.json:836]: `"! ./scripts/should-run-directly.sh || ./scripts/next-start.sh",`

File: scripts/next-build.sh [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+export NVM_DIR="$HOME/.nvm"
+[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"
+source ~/.bashrc
+
+cd next
+
+APP_ENV=tugboat yarn build:preview
+yarn next:start
```
Comment (tjheffner) [inline scripts/next-build.sh:12]: i'd like this moved to a new script file, `next-start.sh`

```
#!/usr/bin/env bash

export NVM_DIR="$HOME/.nvm"
[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"
source ~/.bashrc

cd next

APP_ENV=tugboat yarn start
```

File: .tugboat/config.yml [hunk 5]
```diff
@@ -224,7 +230,7 @@ services:
         - find -L "${DOCROOT}/vendor/va-gov/content-build/node_modules/.bin" -type f -exec chmod +x {} \+
         - find "${DOCROOT}/vendor/va-gov/content-build/script" -type f -exec chmod +x {} \+
 
-        # Build storybook and the frontend in parallel
+        # Build storybook and the frontends in parallel
         - bash -lc 'time task --taskfile=tugboat.yml'
```
Comment (tjheffner) [inline .tugboat/config.yml:235]: one more line needs added here afterwards to start the server

`- bash -lc 'composer va:next:start'`

**Review by tjheffner — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/ap-defect.md [hunk 6]
```diff
@@ -10,13 +10,7 @@ labels: Accelerated Publishing, Defect
 <!-- Describe the problem in a few sentences or less, focusing on the functional issue rather than the technical failure. -->
 
 ### Who is affected?
-<!-- What is the user prevented from doing? About how many users are affected? -->
```
Comment (tjheffner) [inline .github/ISSUE_TEMPLATE/ap-defect.md:13]: i don't think we want any changes to this file in this PR

**Review by tjheffner — CHANGES_REQUESTED** [review]

**Review by tjheffner — APPROVED** [review]

**Pull request merged**